proj_eclipse/src/main/java/smartmag/models/ui/BoxTableModel.java

```diff
@@ -0,0 +1,91 @@
+package smartmag.models.ui;
+
+import java.util.TreeMap;
+
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import javax.swing.table.AbstractTableModel;
+
+import smartmag.data.Box;
+import smartmag.models.BoxModel;
+
+public class BoxTableModel extends AbstractTableModel
+		implements ChangeListener {
+
+	private String[] columnNames;
+	private TreeMap<String, BoxModel> box;
+
+	/**
+	 * costruttore
+	 */
+	public BoxTableModel() {
+		refreshFromModel();
+		this.columnNames = new String[] { "Id", "IDProdotto", "Quantitá" };
+		BoxModel.addChangeListener(this);
+	}
+
+	/**
+	 * aggiorna la lista di box
+	 */
+	public void refreshFromModel() {
+		box = BoxModel.getAllBoxModels();
+		fireTableDataChanged();
+	}
+
+	@Override
+	public int getRowCount() {
+		return box.size();
+	}
+
+	@Override
+	public int getColumnCount() {
+		return columnNames.length;
+	}
+
+	@Override
+	public Object getValueAt(int rowIndex, int columnIndex) {
+		Box b = getBoxModelAt(rowIndex).getBox();
+		return switch (columnIndex) {
+		case 0:
+			yield b.getIndirizzo();
+		case 1:
+			yield b.getProd().getId();
+		case 2:
+			yield b.getQuantità();
+
+		default:
+			throw new IndexOutOfBoundsException("campo inesistente");
+		};
+	}
+
+	/**
+	 * Restituisce l'istanza del modello Box di indice (riga) specificato. Utile
+	 * per ricavare il box selezionato.
+	 * 
+	 * @param index indice posizione box
+	 * @return modello box
+	 */
+	public BoxModel getBoxModelAt(int rowIndex) {
+		BoxModel bm = null;
+		int i = 0;
+
+		for (String id : box.keySet()) {
+			if (i == rowIndex) {
+				bm = box.get(id);
+			}
+			i++;
+		}
+		if (bm == null) {
+			throw new IndexOutOfBoundsException("riga inesistente");
+		}
+
+		return bm;
+	}
+
+	@Override
+	public void stateChanged(ChangeEvent e) {
+		refreshFromModel();
+		fireTableDataChanged();
+	}
+
```

Comment:
Manca l'override del metodo `getColumnName()` per fornire alla tabella le testate delle colonne:

```suggestion
public String getColumnName(int column) {
    return columnNames[column];

}
```

Visto che questo e altri elementi sono comuni a tutti i `TableModel` che andiamo a creare, in futuro possiamo fare refactoring estraendo una superclasse astratta.